projects/igniteui-angular/src/lib/directives/dragdrop/README.md

```diff
@@ -0,0 +1,103 @@
+
+The dragging can be canceled by the user before it starts by setting the `cancel` property of the `dragStart` event to `true`. This will cancel the current dragging logic that could be used.
+
+After the user releases the mouse/touch the drag ghost element is removed from the DOM and if the `hideBaseOnDrag` is enabled it will make the original element visible again and the `dragEnd` event will be emitted. If the `animateOnRelease` input is set to `true` all this will execute after the default animation of the drag ghost is finished which consist of returning it from the last dragged position to the position of the original element. Then the drag ghost will be removed with the rest and the `returnMoveEnd` event will be emitted.
+
+## API
+
+### Inputs
+
+| Name | Type | Default Value | Description |
+| :--- | :--- | :--- | :--- |
+| igxDrag          | any | - | Input used to save data inside the `igxDrag` directive. This can be set when instancing `igxDrag` on an element. |
+| dragTolerance    | number | 5 | Indicates when the drag should start
+     * By default the drag starts after the draggable element is moved by 5px |
+| ghostImageClass  | string | '' | Sets a custom class that will be added to the `dragGhost` element. |
+| hideBaseOnDrag   | boolean | false | Sets if the draggable element should hide when when dragging starts. |
+| animateOnRelease | boolean | false | Enables/disables the draggable element animation when the element is released. |
+
+### Outputs
+
+| Name | Description | Cancelable |
+| :--- | :--- | :--- |
+| dragStart      | Event triggered when the draggable element drag starts. | true |
+| dragEnd        | Event triggered when the draggable element is released. | false |
+| returnMoveEnd  | Event triggered after the draggable element is released and after its animation has finished. | false |
+| dragClicked    | Event triggered when the draggable element is clicked. | false |
+
+### Properties
+
+| Name             | Type        | Description       |
+| :--------------- |:----------- | :---------------- |
+| defaultReturnDuration | string | Sets the default return animation time after dragging occurs. |
+
+<div class="divider--half"></div>
+
+### Methods
+
+| Signature    | Description   |
+| :----------- | :------------ |
+| dropFinished() | Triggers the drop finishing action on the igxDrag when moving it in the DOM from one place to another. This is needed so the igxDrag recalculates it's position once moved in DOM so the current and return animation are based on the new position in DOM. |
+
+#igxDrop
+`igxDrop` directive is used in combination with the `igxDrag` directive to add behavior when element needs to be dropped in an area.
+
+## Usage
+````html
+<div class="cageArea" igxDrop (igxDragEnter)="onDragCageEnter()" (igxDragLeave)="onDragCageLeave()">
```

Comment:
@skrustev "cageArea" -> "dropArea"

Reply:
"onDragCageEnter" => "onDragAreaEnter"
"onDragCageLeave" => "onDragAreaLeave"

Reply:
It is also good to add those event handlers code, so that the user gets the whole picture.